### PR TITLE
MCR-2446 xinclude support

### PR DESCRIPTION
**Pull request by rsteph-de**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2446).

### CODE REVIEW

File: mycore-base/src/main/java/org/mycore/common/xml/MCREntityResolver.java [hunk 1]
```diff
@@ -104,15 +108,29 @@ private static boolean isAbsoluteURL(String url) {
...
+            MCRURIResolver uriResolver = MCRURIResolver.instance();
+            String scheme = uriResolver.getScheme(systemId, null);
+            try {
+                if (scheme != null && uriResolver.getResolver(scheme) != null) {
+                    Source s = MCRURIResolver.instance().resolve(systemId, null, false);
+                    return new MCRSourceContent(s).getInputSource();
+                }
+            } catch (MCRUsageException | TransformerException e) {
+                return null;
+            }
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/common/xml/MCREntityResolver.java]: Simply returning null without any notice? Could cause hard to find errors. We should at least **warn** in the logs.

**Review by yagee-de — CHANGES_REQUESTED** [review]

File: mycore-base/src/main/java/org/mycore/common/xml/MCREntityResolver.java (hunk 1 above)
Comment (rsteph-de) [reply mycore-base/src/main/java/org/mycore/common/xml/MCREntityResolver.java]: O.K. I will add a log message, but the old implementation did also return silent null

**Review by fluetze — COMMENTED** [review]
Are there any security considerations, e.g. including a secret/hidden xml resource from the filesystem into a public webpage?

**Comment by kkrebs** [comment]
we should look into that more closely, so I convert this pull request to draft

**Comment by kkrebs** [comment]
This solution is to generic, we need a seperate ticket

**Pull request closed without merging**